tests/devices_test/dependencies_test.py

```diff
@@ -129,3 +134,111 @@ def test_dependencies_handling(self):
         self.assertFalse(fmt.supported)
         self.assertFalse(fmt.formattable)
         self.assertFalse(fmt.destroyable)
+
+
+class MissingWeakDependenciesTestCase(unittest.TestCase):
+    def setUp(self):
+        self.addCleanup(self._clean_up)
+        self.disk1_file = create_sparse_tempfile("disk1", Size("2GiB"))
+
+    def _clean_up(self):
+        # reload all libblockdev plugins
+        blivet.blockdev.try_reinit(require_plugins=None, reload=False)
+
+        for disk in self.bvt.disks:
+            self.bvt.recursive_remove(disk)
+
+        self.bvt.reset()
+        self.bvt.devicetree.teardown_disk_images()
+        for fn in self.bvt.disk_images.values():
+            if os.path.exists(fn):
+                os.unlink(fn)
+
+    def test_weak_dependencies(self):
+
+        self.bvt = blivet.Blivet()
+        # reinitialize blockdev without the plugins
+        # TODO: uncomment (workaround (1/3) for blivet.reset fail)
+        # blivet.blockdev.try_reinit(require_plugins=[], reload=True)
+
+        self.bvt.disk_images["disk1"] = self.disk1_file
+
+        try:
+            self.bvt.reset()
+        except blivet.blockdev.BlockDevNotImplementedError:  # pylint: disable=catching-non-exception
+            self.fail("Improper handling of missing libblockdev plugin")
+
+        # TODO: remove line (workaround (2/3) for blivet.reset fail)
+        blivet.blockdev.try_reinit(require_plugins=[], reload=True)
+
+        disk1 = self.bvt.devicetree.get_device_by_name("disk1")
+        self.bvt.initialize_disk(disk1)
+        pv = self.bvt.new_partition(size=Size("8GiB"), fmt_type="lvmpv")
+        btrfs = self.bvt.new_partition(size=Size("1GiB"), fmt_type="btrfs")
+
+        # TODO: remove "with" (workaround (3/3) for blivet.reset fail)
+        with patch.object(StorageDevice, "unavailable_dependencies",
+                          new_callable=PropertyMock(return_value=['dummy'])):
+            with six.assertRaisesRegex(self, ValueError, "requires unavailable_dependencies"):
+                self.bvt.create_device(pv)
+            with six.assertRaisesRegex(self, ValueError, "requires unavailable_dependencies"):
+                self.bvt.create_device(btrfs)
+
+        with patch.object(StorageDevice, "unavailable_dependencies",
+                          new_callable=PropertyMock(return_value=set())):
+            try:
+                self.bvt.create_device(pv)
+                self.bvt.create_device(btrfs)
+            except blivet.blockdev.BlockDevNotImplementedError:  # pylint: disable=catching-non-exception
+                self.fail("Improper handling of missing libblockdev plugin")
+
+        try:
```

Comment:
I still don't see what this is testing.

Reply:
Is the purpose is to ensure that you don't get `blockdev.BlockDevNotImplementedError` even when the plugin is missing? Is this the desired behavior?

Reply:
Yes. I think that in case of missing plugin blivet should provide more user friendly exception.
`blockdev.BlockDevNotImplementedError` is generated by `gi.introspection` that haven't found its counterpart. We should provide a better description of the problem to the user.